qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,612 @@
+        coupling_map: CouplingMap,
+        control_flow: bool = False,
+        calibrate_instructions: list[str] | None = None,
+        rng: np.random.Generator = None,
+    ):
+        """
+        Args:
+            num_qubits (int): Number of qubits in the target.
+
+            basis_gates (list[str]): List of basis gate names to be supported by
+                the target. These must be within the supported operations of this
+                target generator, which can be consulted through the
+                ``supported_operations`` property.
+                Common sets of basis gates are ``{"cx", "id", "rz", "sx", "x"}``
+                and ``{"ecr", "id", "rz", "sx", "x"}``. The ``"reset"``,  ``"`delay"``
+                and ``"measure"`` instructions are supported by default, even if
+                not specified via ``basis_gates``.
+
+            coupling_map (CouplingMap): Target's coupling map as an instance of
+                :class:`~.CouplingMap`.
+
+            control_flow (bool): Flag to enable control flow directives on the target
+                (defaults to False).
+
+            calibrate_instructions (list[str] | None): List of instruction names
+                to add default calibration entries to. These must be within the
+                supported operations of this target generator, which can be
+                consulted through the ``supported_operations`` property. If no
+                instructions are provided, the target generator will append
+                empty calibration schedules by default.
+
+            rng (np.random.Generator): Optional fixed-seed generator for default random values.
+        """
+        self._rng = rng if rng else np.random.default_rng(seed=42)
+
+        if num_qubits != coupling_map.size():
+            raise QiskitError(
+                f"The number of qubits (got {num_qubits}) must match "
+                f"the size of the provided coupling map (got {coupling_map.size()})."
+            )
+
+        self._num_qubits = num_qubits
+        self._coupling_map = coupling_map
+
+        # Hardcoded default target attributes. To modify,
+        # access corresponding properties through the public `Target` API
+        super().__init__(
+            description=f"Generic Target with {num_qubits} qubits",
+            num_qubits=num_qubits,
+            dt=0.222e-9,
+            qubit_properties=[
+                QubitProperties(
+                    t1=self._rng.uniform(100e-6, 200e-6),
+                    t2=self._rng.uniform(100e-6, 200e-6),
+                    frequency=self._rng.uniform(5e9, 5.5e9),
+                )
+                for _ in range(num_qubits)
+            ],
+            concurrent_measurements=[list(range(num_qubits))],
+        )
+
+        # Ensure that Reset, Delay and Measure are in
+        # self._basis_gates
+        # so that their instructions are added to the target.
+        self._basis_gates = basis_gates
```

Comment:
I did originally try making it a set, and then ran into lots of issues with flaky unit tests. This is because the instruction noise values are generated iterating over the basis gates, and querying a random generator (with fixed seed) at every iteration. Because sets do not have a fixed order, the gate properties were not generated in the same order every time the unit tests were ran, which caused the gates to have different properties every time, resulting in non-reproducible results even when the seed was fixed. There might be a way of getting the best of both worlds and maybe converting from a set to an ordered list, but at that point it felt not much better than checking for those 3 instructions. Let me know what you think.